homeassistant/components/homekit_controller/config_flow.py

```diff
@@ -73,7 +73,7 @@ def ensure_pin_format(pin):
     match = PIN_FORMAT.search(pin)
     if not match:
         raise aiohomekit.exceptions.MalformedPinError(f"Invalid PIN code f{pin}")
-    return "{}-{}-{}".format(*match.groups())
+    return f"{match.groups()[0]}-{match.groups()[1]}-{match.groups()[2]}"
```

Comment:
With my homekit_controller codeowner hat on I'd prefer you assign `groups()` to a temporary variable rather than call it multiple times. Maybe something like:

```python
left, middle, right = match.groups()
return f"{left}-{middle}-{right}"
```

Or again get rid of the format entirely.

```
"-".join(match.groups())
```

Reply:
The second option looks better in my opinion so I have altered it accordingly :)